include/solvers/vof_algebraic_interface_reinitialization.h

```diff
@@ -0,0 +1,435 @@
+   */
+  GlobalVectorType &
+  get_evaluation_point() override
+  {
+    return this->evaluation_point;
+  }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * local_evaluation_point for the subequation currently solved.
+   *
+   * @return The local evaluation point. Ghosts cells are not considered in
+   * this evaluation.
+   */
+  GlobalVectorType &
+  get_local_evaluation_point() override
+  {
+    return this->local_evaluation_point;
+  }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * newton_update for the subequation currently solved.
+   *
+   * @return The direction used to perform the newton iteration.
+   */
+  GlobalVectorType &
+  get_newton_update() override
+  {
+    return this->newton_update;
+  }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * present_solution for the subequation currently solved.
+   *
+   * @return Vector containing all the values of the solution.
+   */
+  GlobalVectorType &
+  get_present_solution() override
+  {
+    return this->present_solution;
+  }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * system_rhs for the subequation currently solved.
+   *
+   * @return Right-hand side vector.
+   */
+  GlobalVectorType &
+  get_system_rhs() override
+  {
+    return this->system_rhs;
+  }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * nonzero_constraints for the subequation currently solved.
+   *
+   * @return Container of nonzero constraints that arise from several sources such
+   * as boundary conditions and hanging nodes in the mesh. See the deal.II
+   * documentation on constraints on degrees of freedom for more information.
+   */
+  AffineConstraints<double> &
+  get_nonzero_constraints() override
+  {
+    return this->nonzero_constraints;
+  }
+
+
+  /**
+   * @brief Output the \f$L_2\f$ and \f$L_\infty\f$ norms of the correction vector.
+   *
+   * @param[in] display_precision Number of outputted digits.
+   */
+  void
+  output_newton_update_norms(const unsigned int display_precision) override
+  {
+    this->pcout << std::setprecision(display_precision)
+                << "\t||dphi_reinit||_L2 = " << std::setw(6)
+                << this->newton_update.l2_norm() << std::setw(6)
+                << "\t||dphi_reinit||_Linfty = "
+                << std::setprecision(display_precision)
+                << this->newton_update.linfty_norm() << std::endl;
+  }
+
+private:
+  /**
+   * @brief Write parallel VTU files of quantities of interest for
+   * the algebraic interface reinitialization process.
+   *
+   * @param[in] step Integer indicating the reinitialization step number.
+   *
+   * @note Only the reinitialization steps of the last global time-step
+   * (simulation time-step) ran are outputted.
+   */
+  void
+  write_output_results(const unsigned int step);
+
+  /**
+   * @brief Identify minimal cell size.
```

Comment:
I would move this function to our utilties instead. It does not need to be in this class.